metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg_common.py

```diff
@@ -126,9 +130,7 @@ def validate_catalog_size(cls, value):
         return value
 
     def is_profiling_enabled(self) -> bool:
-        return self.profiling.enabled and is_profiling_enabled(
```

Comment:
we could actually use `@functools.lru_cache()` here - and avoid changing the code at all